src/libcore/ops/generator.rs

```diff
@@ -84,6 +84,14 @@ pub trait Generator {
     /// represents a completed future.
     type Return;
 
+    /// The type of argument this generator takes when resuming.
+    ///
+    /// This associated type corresponds to the argument the generator takes,
+    /// and to the type of the `yield` expression inside the generator.
+    // NB: The order of the assoc. types is important here!
+    #[cfg(not(bootstrap))]
+    type Resume;
```

Comment:
Note that having this be an associated type rather than a type parameter to the trait adds significant limitations, probably the most important of which is that it cannot contain any lifetimes that aren't `'static` or built into the type of the generator. I'd expect to see it as a type parameter, or at least a GAT in the final generator design.

I don't think this is worth blocking this PR on, as it's useful to have an implementation here, especially for unblocking no-std futures, but it does seem worthy of a comment.

Reply:
Yeah, that's the approach https://github.com/cavedweller/rust/commit/11c94220a62cd937090e9e6a04e50bb5f434ba6e took, but when I noticed that I was already ~80% finished. I'll open up a follow-up issue for that once this is merged.

Reply:
Turns out this isn't terribly complicated, so I implemented it now.